src/robot.cpp

```diff
@@ -167,9 +167,15 @@ void Robot::WaitUntilReady()
         if (((std::chrono::duration<double>)(std::chrono::system_clock::now() - last)).count() > 0.001)
         {
             last += std::chrono::milliseconds(1);
-            ParseSensorData();
-            SendCommand();
-        } else {
+            if (!IsAckMsgReceived()) {
+                SendInit();
+            } else {
+                ParseSensorData();
+                SendCommand();
```

Comment:
we need to set the joint to 0 torques

Reply:
I have troubles following your comment. Do you mean we have to set zero torques in the `SendInit` function?

Also, we didn't send zero torques before, so I am not sure why we should add it here now?

Reply:
It feels unsafe to call for sendCommand without an obvious set before.
The sendInit is not controlling the robot, it just send metadata if I recall properly.
the sendCommand is sending the desired torques.
So at the beginning of this method I would call `joint->setZeroTorques` and then execute the loop.

Reply:
okay. I've added the `joints->SetZeroCommands();` command.